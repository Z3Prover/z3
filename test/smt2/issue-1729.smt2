; GitHub issue

(assert (forall ((x (_ BitVec 4)))
                (forall ((y (_ BitVec 4)))
                        (= (bvadd x y) #x0))))

(apply (using-params bit-blast :blast-quant true))
