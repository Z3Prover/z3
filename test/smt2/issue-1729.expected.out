(goals
(goal
  (forall ((x.0 Bool) (x.1 Bool) (x.2 Bool) (x.3 Bool))
    (forall ((y.0 Bool) (y.1 Bool) (y.2 Bool) (y.3 Bool))
      (let ((a!1 (= (not x.2) (= y.2 (or (not x.3) (not y.3)))))
            (a!2 (not (or (not x.2) (or (not x.3) (not y.3)))))
            (a!3 (not (or (not y.2) (or (not x.3) (not y.3))))))
      (let ((a!4 (or (not (or (not x.2) (not y.2))) a!2 a!3)))
      (let ((a!5 (or (not (or (not y.1) (not x.1)))
                     (not (or (not x.1) (not a!4)))
                     (not (or (not y.1) (not a!4))))))
      (let ((a!6 (or (= (not x.3) y.3)
                     a!1
                     (= (not x.1) (= (not y.1) a!4))
                     (= (not x.0) (= (not y.0) a!5)))))
        (not a!6)))))))
  :precision precise :depth 1)
)
